accounts/users.go

```diff
@@ -20,54 +21,70 @@ func CheckPasswordHash(hash, password []byte) bool {
 	return err == nil
 }
 
+// ValidatePassword ensures a password meets minimum security requirements
+func ValidatePassword(password string) bool {
+	var containsNum, containsUpper, containsSpecial bool
+	letters := 0
+
+	for _, c := range password {
+		switch {
+		case unicode.IsNumber(c):
+			containsNum = true
+		case unicode.IsUpper(c):
+			containsUpper = true
+			letters++
+		case unicode.IsPunct(c) || unicode.IsSymbol(c):
+			containsSpecial = true
+		case unicode.IsLetter(c) || c == ' ':
+			letters++
+		default:
+			return false
+		}
+	}
+	validLen := letters >= 6 && letters <= 10
+	return containsNum && containsUpper && containsSpecial && validLen
+}
+
 // CreateUser inserts a new user into the database
 func (a *AccountClient) CreateUser(user *User) error {
 
-	// Check if user with that email already exists
 	userExists, err := a.databaseClient.CheckUserExistsByEmail(user.Email)
 	if err != nil {
 		return errors.Wrap(err, "CreateUser: Error checking if user exists")
 	}
 
-	// Do not create allow user to be created if email is already taken
+	// err email already taken
 	if userExists {
 		return ErrEmailMustBeUnique
 	}
 
-	// Hash password
+	// hash password
 	hash, err := HashPassword(user.Password)
 	if err != nil {
 		return errors.Wrap(err, "CreateUser: Error could not hash password")
 	}
 
-	// Capitalize first letter of names
-	firstName := strings.Title(user.FirstName)
-	lastName := strings.Title(user.LastName)
-
-	// Insert new user
-	err = a.databaseClient.InsertUser(user.Email, firstName, lastName, hash)
+	err = a.databaseClient.InsertUser(user.Email, strings.Title(user.FirstName), strings.Title(user.LastName), hash)
 	if err != nil {
 		return errors.Wrap(err, "CreateUser: Error inserting user")
 	}
 
 	return nil
 }
 
-// AuthenticateUser checks if user is valid
+// AuthenticateUser checks if valid user
 func (a *AccountClient) AuthenticateUser(email, password string) (int, error) {
 
-	// Get user by email
 	user, err := a.databaseClient.GetUserByEmail(email)
 	if err != nil {
-		if err.Error() == "Not Found" {
-			return 0, ErrUserNotFound
-		}
 		return 0, err
 	}
+	if user == nil {
```

Comment:
In other areas, you have a new line between if statements on the same block levels. I don't care either way since this is a nitpick.